examples/solid-js/src/todo-list.tsx

```diff
@@ -0,0 +1,63 @@
+import { For, createEffect } from 'solid-js';
+import { Doc, createFireproof } from '@fireproof/solid-js';
+
+// You can have a global database that any Solid component can import
+export const todoList = createFireproof('todo-list');
+
+type Todo = { text: string; date: number; completed: boolean };
+
+export const TodoList = () => {
+  const { database, createDocument, createLiveQuery } = todoList
+  const todos = createLiveQuery<Todo>('date', { limit: 10, descending: true })
+  const [todo, setTodo, saveTodo] = createDocument<Todo>({
+    text: '',
+    date: Date.now(),
+    completed: false,
+  });
```

Comment:
Now that we have generics support, we can just pass in our custom type to the invocations of `createLiveQuery` / `createDocument` and get full type information.